### PR TITLE
feat: Navigate to selected searched message (WPB-4986)

### PR DESCRIPTION
**Pull request by alexandreferris**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4986" title="WPB-4986" target="_blank">SpikeWPB-4986</a>  [Android] Go to the original message in the conversation 
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [X] contains a reference JIRA issue number like `SQPIT-764`
    - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

When clicking on a searched messages result (a message) it wouldn't navigate to the selected message on Conversation Screen.

### Causes (Optional)

Not implemented.

### Solutions

With Kalium's PR we have the selected message position and are passing it as an offset to [GetMessagesForConversationUseCase.kt](https://github.com/wireapp/wire-android-reloaded/pull/2412/files#diff-6497b06a39a62979866de08efc6714d0c5064b2949dc54a6e88a6f9d26e636ac) so it relies correctly on some messages before and after the selected message.

[ConversationNavArgs.kt](https://github.com/wireapp/wire-android-reloaded/pull/2412/files#diff-146367a9d5b5e35020e5bbc473237c37644778edffac9ebf33f7099157c2d183) now holds a new param: `searchedMessageId` so from `SearchConversationMessagesScreen` we can continue going forward on navigation instead of trying to pass back the value through 2 screens.

There is a new param on [SearchBar.kt](https://github.com/wireapp/wire-android-reloaded/pull/2412/files#diff-5817352408041f0e628777475c6e0019b6a688156af14ebe5104c015bc581182) that is `isLoading` where it was added a CircularProgressIndicator on the left side of the `X (close)` button on the far right, so the user knows that the messages are being fetched and NOT stuck.

### Dependencies (Optional)

Needs releases with:

- [X] [Kalium PR](https://github.com/wireapp/kalium/pull/2186)

#### Test Coverage (Optional)

- [X] I have added automated test to this contribution

#### How to Test

- Open App
- Open Conversation Screen
- Open Conversation Details Screen
- Open Search Messages Screen
- Search for some messages:

You will notice the following states (please refer to the images added below):
- Default
- No results
- Results
- Loading

### Notes (Optional)

There are still some issues that will be addressed in another time, some of them  are:
- Performance when selecting a very old message and clicking on navigate to bottom (button on the bottom-right side)
- Self deleting messages still appears in search

### Attachments (Optional)

| Default State | No Results State |
| ----------- | ------------ |
|  Default |  No Results |

| Results State | Loading State |
| ----------- | ------------ |
|  Results |  Loading |

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/usecase/GetMessagesForConversationUseCase.kt [hunk 1]
```diff
@@ -50,11 +50,12 @@ class GetMessagesForConversationUseCase @Inject constructor(
         val pagingConfig = PagingConfig(
             pageSize = PAGE_SIZE,
             prefetchDistance = PREFETCH_DISTANCE,
-            initialLoadSize = max(INITIAL_LOAD_SIZE, lastReadIndex + PREFETCH_DISTANCE)
+            initialLoadSize = INITIAL_LOAD_SIZE
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/usecase/GetMessagesForConversationUseCase.kt:53]: We are making sure that the first load size will always be `50` as per the value on the variable

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/usecase/GetMessagesForConversationUseCase.kt [hunk 2]
```diff
@@ -50,11 +50,12 @@ class GetMessagesForConversationUseCase @Inject constructor(
...
         )
         return getMessages(
             conversationId,
-            pagingConfig = pagingConfig
+            pagingConfig = pagingConfig,
+            startingOffset = max(0, lastReadIndex - PREFETCH_DISTANCE)
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/usecase/GetMessagesForConversationUseCase.kt:58]: Here it is where we really are checking which offset we should start from (either `0` when normally entering the screen, or checking the last read message index position)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt [hunk 3]
```diff
@@ -849,12 +850,15 @@ fun MessageList(
...
             ) {
-                itemsIndexed(lazyPagingMessages, key = { _, uiMessage ->
-                    uiMessage.header.messageId
-                }) { index, message ->
+                items(
+                    count = lazyPagingMessages.itemCount,
+                    key = lazyPagingMessages.itemKey { it.header.messageId },
+                    contentType = lazyPagingMessages.itemContentType { it }
+                ) { index ->
+                    val message: UIMessage? = lazyPagingMessages[index]
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt:858]: This had to be updated from `itemsIndexed` to `items` when updating `paging3`

**Review by MohamadJaara — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [1739](https://10.10.124.134/job/AR-build-pipeline/1739/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(8064338)(8064338)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(8064338)(8064338)-staging-compat.apk)

**Review by mchenani — APPROVED** [review]
🤘

**Pull request merged**